storage/remote/queue_manager.go

```diff
@@ -800,14 +919,33 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue chan sample) {
 			s.droppedOnHardShutdown.Add(uint32(droppedSamples))
 			return
 
+		case <-s.reshard:
+			select {
+			case sample := <-queue:
+				// Capture the remaining (in case) sample from the queue.
```

Comment:
When would this happen, and is it guaranteed to only happen once?

Reply:
I have updated the code. This is to accept the samples that would have entered the queue, between when the resharding signal is sent to `t.shards.reshard` and blocking incoming samples at the time of creation of newQueues.